applications/system-service/systemapi.h

```diff
@@ -211,6 +210,65 @@ class SystemAPI : public APIBase {
         toggleSwipeEnabled((SwipeDirection)direction);
     }
     void toggleSwipeEnabled(SwipeDirection direction){ setSwipeEnabled(direction, !getSwipeEnabled(direction)); }
+    Q_INVOKABLE void setSwipeLength(int direction, int length){
+        int maxLength;
+        if(!hasPermission("system")){
+            return;
+        }
+        if(direction <= SwipeDirection::None || direction > SwipeDirection::Down){
+            qDebug() << "Invalid swipe direction: " << direction;
+            return;
+        }
+        if (direction == SwipeDirection::Up || direction == SwipeDirection::Down){
+            maxLength = 1872;
+        }
+        else{
+            maxLength = 1404;
+        }
+        if (length < 0 || length > maxLength){
+            qDebug() << "Invalid swipe length: " << direction;
+            return;
+        }
+        setSwipeLength((SwipeDirection)direction, length);
+    }
+    Q_INVOKABLE void setAllSwipeLengths(int length){
+        for(short i = Right; i <= Down; i++){
+            setSwipeLength(i, length);
+        }
+    }
+    void setSwipeLength(SwipeDirection direction, int length){
+        if (direction == None){
+            return;
+        }
+        switch(direction){
+            case Left:
+                qDebug() << "Swipe Left Length: " << length;
+                break;
+            case Right:
+                qDebug() << "Swipe Right Length: " << length;
+                break;
+            case Up:
+                qDebug() << "Swipe Up Length: " << length;
+                break;
+            case Down:
+                qDebug() << "Swipe Down Length: " << length;
+                break;
+            default:
+                return;
+        }
+        swipeLengths[direction] = length;
+    }
+    Q_INVOKABLE int getSwipeLength(int direction){
+        if(!hasPermission("system")){
+            return false;
```

Comment:
Since this is returning an `int`, I'd rather use `-1` for the failed return value, and avoid mixing types. Especially since this is going over dbus.
```suggestion
            return -1;
```